feat: Remove automatic wallet pass-through (#1951)
dfinity/sdk: CHANGELOG.adoc

== DFX

=== feat!: Remove the wallet proxy and the --no-wallet flag

Breaking change: Canister commands, except for `dfx canister create`, will make the call directly, rather than via the user's wallet. The `--no-wallet` flag is thus removed from `dfx canister` as its behavior is the default.

When upgrading existing canisters, use the `--wallet` flag in conjunction with `dfx identity get-wallet` in order to restore the old behavior.

> 

Reviewer: We need to spell out in detail what commands developers need to execute in order to update the controllers of their existing canisters.

== DFX

=== feat!: Remove the wallet proxy and the --no-wallet flag

Breaking change: Canister commands, except for `dfx canister create`, will make the call directly, rather than via the user's wallet. The `--no-wallet` flag is thus removed from `dfx canister` as its behavior is the default.

> When working with existing canisters, use the `--wallet` flag in conjunction with `dfx identity get-wallet` in order to restore the old behavior. To upgrade an existing canister to work seamlessly with the new system, the following command should be used (substituting `canister_name` for your own):
> [source, bash]

Reviewer: What does "restore the old behavior" mean?

```suggestion
When working with existing canisters, use the `--wallet` flag in conjunction with `dfx identity get-wallet` in order to restore the old behavior.

You will need to upgrade each of your existing canisters to work with the new system.  To do so, execute the following command for each of your canisters (substituting `canister_name` for your own):
[source, bash]
```

----

> dfx canister --wallet "$(dfx identity get-wallet)" update-settings <canister_name> --add-controller "$(dfx identity get-principal)"

Reviewer: It seems like a lot to ask every developer to do this individually for each of their canisters.

Also, there's a little more to it for canisters that are on the ic network (since the local network is the default):
```
dfx canister --network ic --wallet "$(dfx identity --network ic get-wallet)"
 update-settings <canister_name> --add-controller "$(dfx identity get-principal)"
```
And even more if they use a different identity on the ic that they do for local development:
```
dfx --identity icident canister --network ic
 --wallet "$(dfx --identity icident identity --network ic get-wallet)"
 update-settings <canister_name>
 --add-controller "$(dfx --identity icident identity get-principal)"
```

Can you think of a way to simplify the upgrade process?

Author: Should I include both of those commands? That's a lot to edit in a paste, and they presumably know how the flags work already. Re: simplification, I'm not sure if you're asking for a shell solution or a DFX solution.

Reviewer: Just to follow up after our team meeting: let's make sure to include the specific command to run for mainnet canisters:
```
dfx canister --network ic --wallet "$(dfx identity --network ic get-wallet)"
 update-settings <canister_name> --add-controller "$(dfx identity get-principal)"
```
Also, I think we should call out that `dfx --network ic wallet upgrade` is necessary.

Reviewer: Actually we can streamline it a little bit already, using the `--all` parameter rather than calling once with each canister name:
```
dfx canister --network ic --wallet "$(dfx identity --network ic get-wallet)"
 update-settings --all --add-controller "$(dfx identity get-principal)"
```

Author: Should that be for the first command too?

Reviewer: Yes I think we should specify `--all` in the instructions for both.

== DFX

=== feat!: Remove the wallet proxy and the --no-wallet flag

Breaking change: Canister commands, except for `dfx canister create`, will make the call directly, rather than via the user's wallet. The `--no-wallet` flag is thus removed from `dfx canister` as its behavior is the default.

When working with existing canisters, use the `--wallet` flag in conjunction with `dfx identity get-wallet` in order to restore the old behavior.

> You will need to upgrade each of your existing canisters to work with the new system.  To do so, execute the following command for each of your canisters:
> [source, bash]
> ----
> dfx canister --wallet "$(dfx identity get-wallet)" update-settings --all --add-controller "$(dfx identity get-principal)"
> ----
> Or, if you're upgrading a canister on the IC:
> [source, bash]
> ----
> dfx canister --network ic --wallet "$(dfx identity --network ic get-wallet)" update-settings --all --add-controller "$(dfx identity get-principal)"
> ----

Reviewer: ```suggestion
You will need to upgrade your wallet and each of your existing canisters to work with the new system.  To do so, execute the following in each of your dfx projects:
[source, bash]
----
dfx wallet upgrade
dfx canister --wallet "$(dfx identity get-wallet)" update-settings --all --add-controller "$(dfx identity get-principal)"
----
To upgrade projects that you have deployed to the IC mainnet, execute the following:
[source, bash]
----
dfx wallet --network ic upgrade
dfx canister --network ic --wallet "$(dfx identity --network ic get-wallet)" update-settings --all --add-controller "$(dfx identity get-principal)"
----
```
